[PATCH 3/7] [clang] improve NestedNameSpecifier: test changes

mizvekov opened the pull request:
clang test changes

Third patch in the series, starting at https://github.com/llvm/llvm-project/pull/147835

mizvekov committed:
[PATCH 3/7] [clang] improve NestedNameSpecifier: test changes

clang test changes

Second patch in the series, starting at #147835

ldionne commented on libcxx/test/std/thread/futures/futures.task/futures.task.members/ctor1.verify.cpp:
```diff
@@ -31,7 +31,8 @@ int main(int, char**)
     PT p1{init}; // expected-error {{no matching constructor}}
     PT p2{c_init}; // expected-error {{no matching constructor}}
-    PT p3{std::move(init)}; // expected-error {{no matching constructor for initialization of 'PT' (aka 'packaged_task<A (int, char)>')}}
+    PT p3{std::move(init)};
+    // expected-error-re@-1 {{no matching constructor for initialization of 'PT' (aka '{{(std::)?}}packaged_task<A (int, char)>')}}
```
```suggestion
    PT p3{std::move(init)}; // expected-error-re@-1 {{no matching constructor for initialization of 'PT' (aka {{.+}})}}
```

ldionne commented on libcxx/test/std/numerics/rand/rand.eng/rand.eng.lcong/params.verify.cpp:
```diff
@@ -24,7 +24,7 @@ int main(int, char**)
     std::linear_congruential_engine<T, 0, 1, 1> e3;
     std::linear_congruential_engine<T, 1, 0, 1> e4;
-    // expected-error-re@*:* {{static assertion failed due to requirement 'is_unsigned<int>::value'{{.*}}_UIntType must be unsigned type}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_unsigned<int>::value'{{.*}}_UIntType must be unsigned type}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}_UIntType must be unsigned type}}
```

ldionne commented on libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp:
```diff
@@ -77,5 +77,5 @@ using E = test<std::array<NotALiteral, 1>{}>;
 
 using F = test<std::array<std::string, 2>{}>;
-// expected-error@-1 {{type 'std::array<std::string, 2>' (aka 'array<basic_string<char>, 2>') of non-type template parameter is not a structural type}}
+// expected-error-re@-1 {{type 'std::array<std::string, 2>' (aka '{{(std::)?}}array<basic_string<char>, 2>') of non-type template parameter is not a structural type}}
```
```suggestion
// expected-error-re@-1 {{type {{.+}} of non-type template parameter is not a structural type}}
```

ldionne commented on libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp:
```diff
@@ -45,7 +45,7 @@ using E = test<NotALiteral, 1, std::array<NotALiteral, 1>{}>;
 
 using F = test<std::string, 2, std::array<std::string, 2>{}>;
-// expected-error-re@*:* {{type 'std::array<{{(std::)?}}string, 2U{{L{0,2}.*}}>' {{(\(aka 'array<basic_string<char>, 2UL{0,2}>'\) )?}}of non-type template parameter is not a structural type}}
+// expected-error-re@*:* {{type 'std::array<{{(std::)?}}string, 2U{{L{0,2}.*}}>' {{(\(aka '(std::)?array<basic_string<char>, 2UL{0,2}>'\) )?}}of non-type template parameter is not a structural type}}
```
```suggestion
// expected-error-re@*:* {{type {{.+}} of non-type template parameter is not a structural type}}
```

ldionne commented on libcxx/test/std/containers/associative/multimap/multimap.nonmember/compare.three_way.verify.cpp:
```diff
@@ -25,9 +25,9 @@ int main(int, char**) {
     std::multimap<int, int, std::less<int>, std::allocator<int>> s1;
     std::multimap<int, int, std::less<int>, test_allocator<int>> s2;
-    // expected-error-re@*:* {{static assertion failed due to requirement 'is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Allocator::value_type must be same type as value_type}}
```

ldionne commented on libcxx/test/std/containers/associative/multimap/multimap.nonmember/compare.three_way.verify.cpp:
```diff
@@ -25,9 +25,9 @@ int main(int, char**) {
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
     s1 <=> s2;
-    // expected-error-re@*:* {{static assertion failed due to requirement 'is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Allocator::value_type must be same type as value_type}}
```

ldionne commented on libcxx/test/std/containers/associative/map/map.nonmember/compare.three_way.verify.cpp:
```diff
@@ -25,9 +25,9 @@ int main(int, char**) {
     std::map<int, int, std::less<int>, std::allocator<int>> s1;
     std::map<int, int, std::less<int>, test_allocator<int>> s2;
-    // expected-error-re@*:* {{static assertion failed due to requirement 'is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Allocator::value_type must be same type as value_type}}
```

ldionne commented on libcxx/test/std/containers/associative/map/map.nonmember/compare.three_way.verify.cpp:
```diff
@@ -25,9 +25,9 @@ int main(int, char**) {
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
     s1 <=> s2;
-    // expected-error-re@*:* {{static assertion failed due to requirement 'is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '{{(std::)?}}is_same<int, std::pair<const int, int>>::value'{{.*}}Allocator::value_type must be same type as value_type}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Allocator::value_type must be same type as value_type}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_sub_explicit.verify.cpp:
```diff
@@ -51,12 +51,12 @@ void pointer_to_incomplete_type() {
   {
     volatile std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_sub_explicit.verify.cpp:
```diff
@@ -51,12 +51,12 @@ void pointer_to_incomplete_type() {
   {
     std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_sub.verify.cpp:
```diff
@@ -48,12 +48,12 @@ void pointer_to_incomplete_type() {
   {
     volatile std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_sub.verify.cpp:
```diff
@@ -48,12 +48,12 @@ void pointer_to_incomplete_type() {
   {
     std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_add.verify.cpp:
```diff
@@ -48,12 +48,12 @@ void pointer_to_incomplete_type() {
   {
     volatile std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_add.verify.cpp:
```diff
@@ -48,12 +48,12 @@ void pointer_to_incomplete_type() {
   {
     std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_add_explicit.verify.cpp:
```diff
@@ -51,12 +51,12 @@ void pointer_to_incomplete_type() {
   {
     volatile std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne commented on libcxx/test/libcxx/atomics/atomics.types.operations/atomics.types.operations.req/atomic_fetch_add_explicit.verify.cpp:
```diff
@@ -51,12 +51,12 @@ void pointer_to_incomplete_type() {
   {
     std::atomic<void (*)(int)> fun;
-    // expected-error-re@*:* {{static assertion failed due to requirement '!is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
+    // expected-error-re@*:* {{static assertion failed due to requirement '!{{(std::)?}}is_function<void (int)>::value'{{.*}}Pointer to function isn't allowed}}
```
```suggestion
    // expected-error-re@*:* {{static assertion failed due to requirement {{.+}}Pointer to function isn't allowed}}
```

ldionne reviewed (approved):
I reviewed the libc++ changes. I suggest we take the opportunity to make those more robust to future changes without really reducing their bug-finding power. I made my change suggestions such that you should be able to apply them as a batch with minimal effort on your side.

My approval is only for the libc++ part of the changes and it assumes that you take my comments in batch. Thanks!

mizvekov committed:
[PATCH 4/7] [clang] Improve NestedNameSpecifier: clang-tools-extra changes (#148015)

clang tools extra changes

Fourth and last patch in the series, starting at
https://github.com/llvm/llvm-project/pull/147835
